### PR TITLE
Added a doc with coding advice.

### PR DESCRIPTION
**Pull request by erictune**
@lavalamp I cited your Gist

Cites the bash guide mentioned in #1266

### CODE REVIEW

File: docs/devel/coding-conventions.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Coding style advice for contributors
+  - Bash
+    - https://google-styleguide.googlecode.com/svn/trunk/shell.xml
+  - Go
+    - https://github.com/golang/go/wiki/CodeReviewComments
+    - https://gist.github.com/lavalamp/4bd23295a9f32706a48f
```
Comment (lavalamp) [inline docs/devel/coding-conventions.md]: I'm flattered-- but not really style advice, is it? :)

Comment (erictune) [inline docs/devel/coding-conventions.md]: not introducing bugs is always stylish.

Comment (lavalamp) [inline docs/devel/coding-conventions.md]: Haha :)

**Comment by lavalamp** [comment]
LGTM

**Pull request merged**